nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/main/java/org/apache/nifi/processors/elasticsearch/JsonQueryElasticsearch.java

```diff
@@ -0,0 +1,326 @@
+            .name("el-rest-query")
+            .displayName("Query")
+            .description("A query in JSON syntax, not Lucene syntax. Ex: " +
+                    "{\n" +
+                    "\t\"query\": {\n" +
+                    "\t\t\"match\": {\n" +
+                    "\t\t\t\"name\": \"John Smith\"\n" +
+                    "\t\t}\n" +
+                    "\t}\n" +
+                    "}")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor QUERY_ATTRIBUTE = new PropertyDescriptor.Builder()
+            .name("el-query-attribute")
+            .displayName("Query Attribute")
+            .description("If set, the executed query will be set on each result flowfile in the specified attribute.")
+            .expressionLanguageSupported(true)
+            .addValidator(Validator.VALID)
+            .required(false)
+            .build();
+
+    public static final AllowableValue SPLIT_UP_YES = new AllowableValue(
+        "splitUp-yes",
+        "Yes",
+        "Split up results."
+    );
+    public static final AllowableValue SPLIT_UP_HITS_NO = new AllowableValue(
+        "splitUp-no",
+        "No",
+        "Don't split up results."
+    );
+
+    public static final PropertyDescriptor SPLIT_UP_HITS = new PropertyDescriptor.Builder()
+            .name("el-rest-split-up-hits")
+            .displayName("Split up search results")
+            .description("Split up search results into one flowfile per result.")
+            .allowableValues(SPLIT_UP_HITS_NO, SPLIT_UP_YES)
+            .defaultValue(SPLIT_UP_HITS_NO.getValue())
+            .required(true)
+            .expressionLanguageSupported(false)
+            .build();
+    public static final PropertyDescriptor SPLIT_UP_AGGREGATIONS = new PropertyDescriptor.Builder()
+            .name("el-rest-split-up-aggregations")
+            .displayName("Split up aggregation results")
+            .description("Split up aggregation results into one flowfile per result.")
+            .allowableValues(SPLIT_UP_HITS_NO, SPLIT_UP_YES)
+            .defaultValue(SPLIT_UP_HITS_NO.getValue())
+            .required(true)
+            .expressionLanguageSupported(false)
+            .build();
+
+    public static final PropertyDescriptor CLIENT_SERVICE = new PropertyDescriptor.Builder()
+            .name("el-rest-client-service")
+            .displayName("Client Service")
+            .description("An ElasticSearch client service to use for running queries.")
+            .identifiesControllerService(ElasticSearchClientService.class)
+            .required(true)
+            .build();
+
+    private static final Set<Relationship> relationships;
+    private static final List<PropertyDescriptor> propertyDescriptors;
+
+    private ElasticSearchClientService clientService;
```

Comment:
This should be set as volatile since it is referenced by multiple different threads.

Reply:
Done.